src/sentry/api/endpoints/organization_fork.py

```diff
@@ -0,0 +1,158 @@
+import logging
+from string import Template
+
+from django.db import router
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, region_silo_endpoint
+from sentry.api.endpoints.relocations.index import (
+    get_autopause_value,
+    validate_relocation_uniqueness,
+)
+from sentry.api.permissions import SuperuserOrStaffFeatureFlaggedPermission
+from sentry.api.serializers import serialize
+from sentry.hybridcloud.services.organization_mapping import organization_mapping_service
+from sentry.models.relocation import Relocation
+from sentry.organizations.services.organization import organization_service
+from sentry.tasks.relocation import uploading_start
+from sentry.types.region import get_local_region
+from sentry.utils.db import atomic_transaction
+
+ERR_DUPLICATE_ORGANIZATION_FORK = Template(
+    "This organization is already in the process of being forked, relocation id: $uuid"
+)
+ERR_ORGANIZATION_NOT_FOUND = Template("The target organization `$pointer` could not be found.")
+ERR_ORGANIZATION_MAPPING_NOT_FOUND = Template(
+    "The target organization `$slug` has no region mapping."
+)
+ERR_CANNOT_FORK_INTO_SAME_REGION = Template(
+    "The organization already lives in region `$region`, so it cannot be forked into that region."
+)
+
+logger = logging.getLogger(__name__)
+
+
+@region_silo_endpoint
+class OrganizationForkEndpoint(Endpoint):
+    owner = ApiOwner.OPEN_SOURCE
+    publish_status = {
+        "POST": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (SuperuserOrStaffFeatureFlaggedPermission,)
+
+    def post(self, request: Request, organization_id_or_slug) -> Response:
+        """
+        Duplicate an organization across regions. The old organization remains untouched. We kick
+        off the standard `SAAS_TO_SAAS` relocation flow to create a duplicate in this region.
+
+        Because each region silo of the API has its own version of this endpoint, we assume that the
+        target region for the fork is the owning region of the API on which this was called. For
+        example, if we call this endpoint at `us.sentry.io`, we are implicitly saying we would like
+        the target organization forked INTO the `us` region.
+        ``````````````````````````````````````````````````
+
+        :pparam string org_slug: the id or slug of the organization
+
+        :auth: required
+        """
+
+        logger.info("relocations.fork.post.start", extra={"caller": request.user.id})
+
+        org_retrieval_args = {
+            "only_visible": True,
+            "include_projects": False,
+            "include_teams": False,
+        }
+        org_context = (
+            organization_service.get_organization_by_id(id=organization_id_or_slug)
```

Comment:
Will this break if org slug is a number?

Reply:
Valid org slugs cannot be a number:

https://github.com/getsentry/sentry/blob/d152e91d8ec0ed1a2816033ad87c9853ed0e0d40/src/sentry/slug/patterns.py#L12

This is checked upstream by the endpoint resolving regex, so if we're hitting this line of code, we can be sure that numerical == id, non-numerical == slug. This is also the check we use in other endpoints that take org ids or slugs interchangeably.